src/middlewared/middlewared/plugins/ntp_/client.py

```diff
@@ -0,0 +1,49 @@
+import dataclasses
+import socket
+import struct
+import time
+
+
+def format_ntp_packet(data):
+    # unpack the response
+    unpacked = struct.unpack('!B B B b 11I', data)
+
+    recv_timestamp = unpacked[11] + float(unpacked[12]) / 2 ** 32
+    ntp_time = recv_timestamp - 2208988800  # NTP timestamp starts from 1st January 1900
+    recv_utc_time = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(ntp_time))
+    return {
+        'leap': unpacked[0] >> 6 & 0x7,
+        'version': unpacked[0] >> 3 & 0x7,
+        'mode': unpacked[0] & 0x7,
+        'stratum': unpacked[1],
+        'poll': unpacked[2],
+        'precision': unpacked[3],
+        'root_delay': float(unpacked[4]) / 2 ** 16,
+        'root_dispersion': float(unpacked[5]) / 2 ** 16,
+        'ref_id': unpacked[6],
+        'ref_timestamp': unpacked[7] + float(unpacked[8]) / 2 ** 32,
+        'orig_timestamp': unpacked[9] + float(unpacked[10]) / 2 ** 32,
+        'recv_timestamp': recv_timestamp,
+        'recv_timestamp_formatted': recv_utc_time,
+        'tx_timestamp': unpacked[13] + float(unpacked[14]) / 2 ** 32,
+    }
+
+
+@dataclasses.dataclass(slots=True)
+class NTPClient:
+    host: str
+    timeout: int = 1  # second
```

Comment:
Maybe make the `timeout` a default parameter to `make_request`, just in case 1 second isn't enough.

Reply:
When you instantiate the `NTPClient` you can pass a `timeout` argument to it to override it. `make_request` will reference `self.timeout` accordingly.